src/main/java/com/store/book/validation/FieldMatchValidator.java

```diff
@@ -0,0 +1,28 @@
+package com.store.book.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.Objects;
+import org.apache.commons.beanutils.BeanUtils;
+
+public class FieldMatchValidator implements ConstraintValidator<FieldMatch, Object> {
+    private String firstFieldName;
+    private String secondFieldName;
+
+    public void initialize(FieldMatch constraint) {
+        firstFieldName = constraint.first();
+        secondFieldName = constraint.second();
+    }
+
+    public boolean isValid(Object value, ConstraintValidatorContext context) {
+        String firstObj;
+        String secondObj;
+        try {
+            firstObj = BeanUtils.getProperty(value, firstFieldName);
+            secondObj = BeanUtils.getProperty(value, secondFieldName);
+        } catch (Exception e) {
+            return false;
+        }
+        return Objects.equals(firstObj, secondObj);
```

Comment:
you can simplify this method (example)
```suggestion
        Object field = new BeanWrapperImpl(value).getPropertyValue(this.field);
        Object fieldMatch = new BeanWrapperImpl(value).getPropertyValue(this.fieldMatch);
        return Objects.equals(field, fieldMatch);
```